### PR TITLE
Prove that Instance.equals() still doesn't work and fix it

### PR DESCRIPTION
**Pull request by MrVoltz**
May need to check if it works with Mongo and Postgres, will wait for Travis CI result.

### CODE REVIEW

File: lib/instance.js [hunk 1]
```diff
@@ -1012,27 +1012,25 @@ Instance.prototype.decrement = function(fields, options) {
...
-  Utils._.each(this.dataValues, function(value, key) {
-    if (Utils._.isDate(value) && Utils._.isDate(other.dataValues[key])) {
-      result = result && (value.getTime() === other.dataValues[key].getTime());
-    } else {
-      result = result && (value === other.dataValues[key]);
-    }
-  });
+  if (this.get(this.Model.primaryKeyAttribute, {raw: true}) !== other.get(this.Model.primaryKeyAttribute, {raw: true})) {
+    return false;
+  }
```
Comment (mickhansen) [inline lib/instance.js]: We should probably account for multiple primary keys here.

**Comment by janmeier** [comment]
I guess this is technically a breaking change - some people might use the method to check whether a row has changed (even though it would be a weird usage, it would break). Coming from other languages (java for example), you would also expect that an `equals` method checks all attributes.

Should we tag this as 4.0, or write it of as a bug?

**Comment by janmeier** [comment]
Just to be clear, I really think we need this method, since it could replace a lot of internal calls checking if `x.id == y.id` - but maybe it should be called `primaryEquals` or something similar until 4.0

**Comment by janmeier** [comment]
What do you think @mickhansen, bug or 4.0?

**Comment by mickhansen** [comment]
Not at all sure what people usually use `equals` for.
It's rather important whether or not people use it to match rows, or use it to match values.

**Pull request merged**